src/pages/product/[slug].tsx

```diff
@@ -1,14 +1,22 @@
+import { useState } from "react";
 import { useRouter } from "next/router";
-import { Container, Flex, Image, Stack, Text } from "@chakra-ui/react";
+import { Container, Flex, Image, Progress, Stack, Text } from "@chakra-ui/react";
 
 import Empty from "~/components/Empty";
 import ProductContent from "~/components/ProductContent";
 import { allProducts } from "~/data/data";
 
 const ProductPage = () => {
   const { query } = useRouter();
+  const [loading, setLoading] = useState(true);
   const product = allProducts.find((result) => result.slug === query.slug);
 
+  setTimeout(() => {
+    setLoading(false);
+  }, 500);
```

Comment:
you would need to wrap this in useEffect to prevent it from running multiple times
```suggestion
  useEffect(() => {
    const timerId = setTimeout(() => {
      setLoading(false);
    }, 500);
    return () => clearTimeout(timerId);
  }, []);
```